### ISSUE TITLE
Quick wake+sleep sequence can disable frontlight

### ISSUE THREAD

**Issue by fbriere** [koreader/koreader#12246]
* KOReader version: 2024.07
* Devices: Kobo Libra & Forma

#### Issue

Putting the device back to sleep immediately after waking up, before the frontlight has had time to turn on, can result in said frontlight being disabled afterwards.

#### Steps to reproduce

Assuming the frontlight is initially enabled:

- Put the device to sleep and let it rest for a little while (say, 30+ seconds);
- Using either the power button or a sleepcover, wake up the device and immediately put it back to sleep (say, within ½ a second);
- Wake the device again, the frontlight should now be disabled.

(The time intervals are mere guesstimates that seemed to work for me.  YMMV.)

#### Affected versions

This behavior was not present in 2023.04 and appeared in 2023.05.1.  (@zwim's #10305 seems like a prime suspect, which could mean this only applies to Kobo devices.)

#### Further justification

This may sound like a very specific nit to pick, but it's actually something that I'm personally very likely to encounter: I upgraded (from 2023.04 to 2024.07) only a few days ago, and already ran into this a couple of times since.  Basically, I'll start to open the sleepcover, my thumb will slip, and the cover will close back, thus disabling the frontlight.  This can be disorienting at first, especially in a dark room where one has to turn on another light to see what happened.  (Now that I know what's going on, I only need to tap the bottom-left corner once in a while, which makes this easier to deal with — although not less annoying.  :smile:)

**Comment by NiLuJe** [comment]
Verbose debug logs, please.

(I've only ever been able to reproduce something possibly similar by doing some *very* wonky shit, FWIW).

**Comment by NiLuJe** [comment]
Also, because it probably matters:

> Wake the device again, the frontlight should now be disabled.

Wake it *how*? Sleep cover or power button? In both case: is the cover closed or open at the time?

**Comment by NiLuJe** [comment]
> This behavior was not present in 2023.04 and appeared in 2023.05.1

Did you thoroughly test that (i.e., did you actually run 2023.05.1 for any length of time?).

----

(FWIW, it is most likely Kobo specific, if for no other reason that, one, frontlight handling is a mess, especially on Kobo, and, two, Kobo is the only platform with extra sleep-cover logic).

**Comment by NiLuJe** [comment]
> (@zwim's #10305 seems like a prime suspect, which could mean this only applies to Kobo devices.)

So we don't get his ears burning for no good reason, IIRC, I probably wrote most of the offending logic (i.e., the not-maths in themaking-it-fit-into-powerd-awful-api bits) course of that PR.

Might also actually be less of a contender than https://github.com/koreader/koreader/pull/10426, which falls in the same version range, and that I wrote all on my own ;p.

**Comment by NiLuJe** [comment]
Might be device-specific to some-extent, but can confirm on the Forma.

* Initial delay is a red-herring (was suspicious because that could roughly match the delay until we actually enter suspend).
* SleepCover is a red-herring (glad I spent 5min finding the magnet -_-". FWIW, it's around the third row in the FM, right in the middle of the vertical axis).

TL;DR:

1. Press power button.
2. Wait for screensaver & frontlight to turn-off.
3. Press power button.
4. Quickly press power button again.
5. Wait a few seconds, press power button: wakes up w/ fl off.

**Comment by spinktvis** [comment]
I have the same issue in the Kobo Libra Colour and was about to create a new ticket for it.

Is easily reproducable with the sleep cover by opening it slightly and closing it again a few times.

**Comment by jospalau** [comment]
It does not work in my Kobo Libra Colour after installing the last nightly. Same as @spinktvis mentioned in previous post. Anyone can confirm the fix works for this device?

**Comment by NiLuJe** [comment]
Define "latest nightly" ;).

**Comment by jospalau** [comment]
Development update

**Comment by NiLuJe** [comment]
I meant I need the actual *version* ;).

Keep in mind I can't reproduce on a Libra Colour myself,; even before this PR.

**Comment by jospalau** [comment]
Got it :)

v2024.07-49-g9b9898d46_2024-08-01

Just trying to confirm of for the other guys having this model works to discard it is something mangled in my insulation

**Comment by jospalau** [comment]
Yes, as @spinktvis mentioned it happens with the sleep cover by opening it slightly and closing it again a few times.

**Comment by NiLuJe** [comment]
> I can't reproduce on a Libra Colour myself

Oh, that's probably because I had "keep screensaver displayed until tap" enabled.

I see a similar behavior as on the Forma without it (i.e., it works after this PR ;p).

**Comment by NiLuJe** [comment]
Okay, that one's more likely sleepcover related, so I'm going to need verbose debug logs.

**Comment by NiLuJe** [comment]
(For future reference: magnet is on the edge of the screen closest to the button, right between the two buttons).

**Comment by fbriere** [comment]
FWIW, both `2024.07`+#12256 and `2024.07-49` fixed the issue for me (Libra/Forma, button or cover).  Thanks @NiLuJe!

**Comment by spinktvis** [comment]
> Okay, that one's more likely sleepcover related, so I'm going to need verbose debug logs.
> 
> (i.e., I need to confuse the magnet real bad to get something wonky to happen, which is fairly easy when you're actually _looking_ for the damn thing with a fridge magnet ;p)

Attached are the debug logs, reproduced it a few times to be sure.
[crash.log](https://github.com/user-attachments/files/16481080/crash.log)
